pkg/kubernetes/scripts/app.js

```diff
@@ -210,8 +209,10 @@
 
             var promise = discoverSettings().then(function(settings) {
                 var ret = [];
-                if (settings.flavor === "openshift")
+                if (settings.flavor === "openshift") {
+                    ret.push(loader.watch("projects", $rootScope));
```

Comment:
If the watch fails the loader loads the projects and the functionality should be the same as it is currently.

Having the watch just better helps the UI pick up changes that we were sometimes missing before and hopefully will help clean up some of the flakes.